index.js

```diff
@@ -7,12 +7,55 @@
 
 'use strict';
 
-module.exports = function dashify(str) {
-  if (typeof str !== 'string') {
+module.exports = dashify
+
+function dashify(string) {
+  if (typeof string !== 'string') {
     throw new TypeError('expected a string');
   }
-  str = str.replace(/([a-z])([A-Z])/g, '$1-$2');
-  str = str.replace(/[ \t\W]/g, '-');
-  str = str.replace(/^-+|-+$/g, '');
-  return str.toLowerCase();
+
+  var sentence = []
+  var needsDash = false
+
+  for(var i=0; i<string.length; i++) {
```

Comment:
Got 'em! I don't use semi-colons unless I'm starting a line with an array/object literal, so I tried to add them in, but missed a few. :)